### DISCUSSION TITLE
Remove dependency on jQuery

### DISCUSSION THREAD

**Discussion by JD557** [scala/scala3#18903] [General Discussion]
Right now, ScalaDoc depends on jQuery.

From what I can tell, the main reason for it is the table of contents plugin [toc](https://github.com/lampepfl/dotty/blob/8cb49457adb2bc014d90e781ef664ae57d1d2d86/docs/_spec/public/scripts/toc.js).

I would like to suggest replacing jQuery and toc with vanilla JavaScript implementations. While jQuery is still maintained, most of the features are now part of the JavaScript spec, so the migration is not that painful.

On top of that, the current way things are setup doesn't play nicely with JS blockers: Scaladoc works with both JS enabled and disabled, but breaks if JS is enabled only for the page domain (and not for the CDNs)

**Comment by SethTisue** [comment]
Seems likely a PR on this would be accepted.

**Comment by SethTisue** [comment]
seems related? https://github.com/lampepfl/dotty/issues/19208

**Comment by Math-ias** [comment]
I was once again looking for beginner friendly issues to contribute to. I have changes to accomplish just this! Should I create a PR referencing the discussion or does it need to link to an issue first? Here are my changes. https://github.com/Math-ias/scala3-fork/tree/discussion-18903

**Reply by sjrd** [reply]
Feel free to open a PR.
